torch/testing/_internal/common_nn.py

```diff
@@ -4889,12 +4949,36 @@ def __init__(self, *args, **kwargs):
     def _do_test(self, test_case, module, input):
         num_threads = torch.get_num_threads()
         torch.set_num_threads(1)
-        test_case.check_jacobian(module, input, self.jacobian_input)
+        input_tuple = input if isinstance(input, tuple) else (input,)
+        test_case.check_jacobian(module, input_tuple, self.jacobian_input)
         if self.check_gradgrad:
+            # this function creates a lambda function based on the number of tensors in the input
+            # eg. if input is (tensor1, tensor2), the generated lambda function will be:
+            #     lambda _0, _1, *args, **kw: test_case._forward(module, (_0, _1, ))
+            def get_lambda_func(inputs, test_case, module):
+                # we assume the inputs is always not empty
+                len_inputs = len(inputs)
+                if len_inputs == 1:
+                    return lambda _0, *args, **kw: test_case._forward(module, (_0, ))
+                if len_inputs == 2:
+                    return lambda _0, _1, *args, **kw: test_case._forward(module, (_0, _1, ))
+                # generate lambda function for case that length of input > 2
+                helper_str = "def _get_lambda_func(test_case, module):"
+                fn_str = "return lambda "
+                in_str = ""
+                for i in range(0, len(inputs)):
+                    in_str = in_str + "_" + str(i) + ", "
+                fn_str = fn_str + in_str + "*args, **kw: test_case._forward(module, (" + in_str + "))"
+                helper_str = helper_str + fn_str
+                # we have to do exec to define a function in order to pass test_case and module in
+                tmp_locals = {}
+                exec(helper_str, {}, tmp_locals)
+                return tmp_locals['_get_lambda_func'](test_case, module)
+
             # could probably unify check_jacobian above with this.
             params = tuple(x for x in module.parameters())
-            _assertGradAndGradgradChecks(test_case,
-                                         lambda x, *args, **kw: test_case._forward(module, x), (input,) + params)
+            _assertGradAndGradgradChecks(
+                test_case, get_lambda_func(input_tuple, test_case, module), input_tuple + params)
```

Comment:
Can't this just be:
```
num_inputs = len(input_tuple)
_assertGradAndGradgradChecks(test_case, lambda *args: test_case._forward(module, args[num_inputs]), input_tuple + params)
```
?

Reply:
I will do a test. I think you are right!. I am so used to C++ placeholder thing......., will update this later.